src/js/FileSystem.js

```diff
@@ -0,0 +1,85 @@
+
+class FileSystem {
+
+    _createFile(fileHandle) {
+        return {
+            name: fileHandle.name,
+            _fileHandle: fileHandle,
+        };
+    }
+
+    async pickSaveFile(suggestedName, description, extension) {
+        const fileHandle = await window.showSaveFilePicker({
+            suggestedName: suggestedName,
+            types: [{
+                description: description,
+                accept: {
+                    "application/unknown": [extension],
+                },
+            }],
+        });
+
+        const file = this._createFile(fileHandle);
+
+        if (await this.verifyPermission(file, true)) {
+            return file;
+        }
+    }
+
+    async pickReadFile(description, extension) {
```

Comment:
Suggest using Open instead of Read

```suggestion
    async pickOpenFile(description, extension) {
```

Reply:
Ok, I will change...